TrafficCapture/trafficReplayer/src/test/java/org/opensearch/migrations/replay/TestHttpServerContext.java

```diff
@@ -0,0 +1,44 @@
+package org.opensearch.migrations.replay;
+
+import org.opensearch.migrations.testutils.HttpFirstLine;
+import org.opensearch.migrations.testutils.SimpleHttpResponse;
+
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.Map;
+
+public class TestHttpServerContext {
+
+    static String getUriForIthRequest(int i) {
+        return String.format("/%04d", i);
+    }
+
+    static String getRequestString(int i) {
+        return String.format("GET %s HTTP/1.1\r\n" +
+                        "Connection: Keep-Alive\r\n" +
+                        "Host: localhost\r\n" +
+                        "User-Agent: UnitTest\r\n" +
+                        "Connection: Keep-Alive\r\n" +
+                        "\r\n",
+                getUriForIthRequest(i));
+    }
+
+    public static String SERVER_RESPONSE_BODY_PREFIX = "Boring Response to ";
+    public static Duration SERVER_RESPONSE_LATENCY = Duration.ofMillis(100);
+
+    public static SimpleHttpResponse makeResponse(HttpFirstLine r) {
+        try {
+            Thread.sleep(SERVER_RESPONSE_LATENCY.toMillis());
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
```

Comment:
Why is this a RuntimeException?

Reply:
Because it isn't checked.  Checked exceptions can't be thrown from many of Java's function interfaces (including java.util.Function, which this function is cast to elsewhere in the code)